android/src/org/ppsspp/ppsspp/NativeActivity.java

```diff
@@ -157,6 +158,30 @@ public void GetScreenSize(Point size) {
 		}
 	}
 
+	public static final int REQUEST_CODE_STORAGE_PERMISSION = 1337;
+
+	@TargetApi(23)
+	public void askForStoragePermission() {
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+			if (this.checkSelfPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
```

Comment:
We really need to check the permission status on start, so that if they've already granted it, we don't ask them every time.

-[Unknown]